Sources/ComposableArchitecture/Reducer.swift

```diff
@@ -238,7 +249,7 @@ public struct Reducer<State, Action, Environment> {
       guard let (id, localAction) = toLocalAction.extract(from: globalAction) else { return .none }
       return self.optional
         .reducer(
-          &globalState[keyPath: toLocalState][id],
+          &globalState[keyPath: toLocalState][id: id],
```

Comment:
I've updated the interface to take an explicit `id` parameter to avoid any ambiguity for `Int`-based ids.